src/server/enterprise/testing/cmd_test.go

```diff
@@ -42,3 +45,192 @@ func TestRegisterPachd(t *testing.T) {
+		pachctl auth get-robot-token --enterprise -q {{.alice}} | pachctl auth use-auth-token --enterprise
+		pachctl auth get-robot-token -q {{.bob}} | pachctl auth use-auth-token
+		pachctl auth whoami --enterprise | match {{.alice}}
+		pachctl auth whoami | match {{.bob}}
+		`,
+		"id", tu.UniqueString("cluster"),
+		"license", tu.GetTestEnterpriseCode(t),
+		"token", tu.RootToken,
+		"alice", tu.UniqueString("alice"),
+		"bob", tu.UniqueString("bob"),
+	).Run())
+}
+
+// TestConfig tests getting and setting OIDC configuration for the identity server
+func TestConfig(t *testing.T) {
+	resetClusterState(t)
+	defer resetClusterState(t)
+
+	require.NoError(t, tu.BashCmd(`
+		echo {{.license}} | pachctl enterprise activate
+		echo {{.token}} | pachctl auth activate --enterprise --issuer http://pach-enterprise.enterprise:658 --supply-root-token
+		pachctl enterprise register --id {{.id}} --enterprise-server-address pach-enterprise.enterprise:650 --pachd-address pachd.default:650
+		echo {{.token}} | pachctl auth activate --supply-root-token --client-id pachd2
+			`,
+		"id", tu.UniqueString("cluster"),
+		"token", tu.RootToken,
+		"license", tu.GetTestEnterpriseCode(t),
+	).Run())
+
+	require.NoError(t, tu.BashCmd(`
+		pachctl auth set-config --enterprise <<EOF
+{
+	"issuer": "http://pach-enterprise.enterprise:658",
+        "localhost_issuer": true,
+	"client_id": localhost,
+	"redirect_uri": "http://pach-enterprise.enterprise:650"
+}
+EOF
+	`).Run())
+
+	require.NoError(t, tu.BashCmd(`
+		pachctl auth get-config --enterprise \
+		  | match '"issuer": "http://pach-enterprise.enterprise:658"' \
+		  | match '"localhost_issuer": true' \
+		  | match '"client_id": "localhost"' \
+		  | match '"redirect_uri": "http://pach-enterprise.enterprise:650"'
+		`,
+	).Run())
+}
+
+// TestLoginEnterprise tests logging in to the enterprise server
+func TestLoginEnterprise(t *testing.T) {
+	resetClusterState(t)
+	defer resetClusterState(t)
+
+	ec, err := client.NewEnterpriseClientForTest()
+	require.NoError(t, err)
+
+	require.NoError(t, tu.BashCmd(`
+		echo {{.license}} | pachctl enterprise activate
+		echo {{.token}} | pachctl auth activate --enterprise --issuer http://pach-enterprise.enterprise:658 --supply-root-token
+		pachctl enterprise register --id {{.id}} --enterprise-server-address pach-enterprise.enterprise:650 --pachd-address pachd.default:650
+		echo {{.token}} | pachctl auth activate --supply-root-token --client-id pachd2
+		echo '{"username": "admin", "password": "password"}' | pachctl idp create-connector --id test --type mockPassword --name test  --config -
+		`,
+		"id", tu.UniqueString("cluster"),
+		"token", tu.RootToken,
+		"license", tu.GetTestEnterpriseCode(t),
+	).Run())
+
+	cmd := exec.Command("pachctl", "auth", "login", "--no-browser", "--enterprise")
+	out, err := cmd.StdoutPipe()
+	require.NoError(t, err)
+
+	require.NoError(t, cmd.Start())
+	sc := bufio.NewScanner(out)
+	for sc.Scan() {
```

Comment:
Any particular reason for the scanner instead of just reading it all into a buffer?
https://golang.org/pkg/os/exec/#Cmd.Output

Reply:
The command blocks until the user has completed the oauth flow in the browser, we use a scanner to get the output line-by-line and check for the line with the URL to visit.